src/main/java/io/jenkins/plugins/mcp/server/extensions/DefaultMcpServer.java

```diff
@@ -93,8 +120,7 @@ public boolean triggerBuild(
 
         if (job != null) {
             job.checkPermission(Item.BUILD);
-            Cause.UserIdCause userIdCause = new Cause.UserIdCause();
-            CauseAction action = new CauseAction(userIdCause);
+            CauseAction action = new CauseAction(new MCPCause());
```

Comment:
CauseAction accepts multiple causes.
You can add both UserId and new mcp causes.